### PR TITLE
Add extra validations by checking current userId, execute su, other p…

### PR DESCRIPTION
**Pull request by youran-studio**
…ossible path of which command.

### CODE REVIEW

File: android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java [hunk 1]
```diff
@@ -40,5 +50,23 @@ private boolean checkRootMethod2() {
         } finally {
             if (process != null) process.destroy();
         }
+    }       
+
+    private boolean checkUserIdAndExecuteSu() {
```
Comment (hdsenevi) [inline android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java:55]: Where is this code referred from?

File: android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java [hunk 2]
```diff
@@ -31,7 +31,17 @@ private boolean checkRootMethod1() {
...
+            final String whichPathNew = "system/bin/which";
+
+            if (new File(whichPathOld).exists()) {
+                process = Runtime.getRuntime().exec(new String[]{whichPathOld, "su"});
+            } else if (new File(whichPathNew).exists()) {
+                process = Runtime.getRuntime().exec(new String[]{whichPathNew, "su"});
+            } else {
+                process = Runtime.getRuntime().exec(new String[]{"which", "su"});
+            }
+
```
Comment (hdsenevi) [inline android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java:44]: Don't we need to do a similar kind of thing on `LessThan23.java` as well (https://github.com/GantMan/jail-monkey/blob/master/android/src/main/java/com/gantix/JailMonkey/Rooted/LessThan23.java)

**Review by hdsenevi — CHANGES_REQUESTED** [review]
Few minor comments that needs addressing. Otherwise, looking good.

File: android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java (hunk 2 above)
Comment (youran-studio) [reply android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java:44]: Plum doesn't support less than 23 I think.

File: android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java (hunk 1 above)
Comment (youran-studio) [reply android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java:55]: So I wrote the code.   
Does it answer the question?

File: android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java [hunk 3]
```diff
@@ -40,5 +50,23 @@ private boolean checkRootMethod2() {
...
+    private boolean checkUserIdAndExecuteSu() {
+        Process process = null;
+
+        try {
+            process = Runtime.getRuntime().exec("id");
+            BufferedReader in = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            if (in.readLine().contains("uid=0")) return true;
+
+            // Android Will throw exception if permission denied when user try to invoke Su
+            process = Runtime.getRuntime().exec("su");
```
Comment (youran-studio) [inline android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java:64]: Tested In android platform, pf will throw exception while executing SU on a non-rooted devices.

**Review by hdsenevi — APPROVED** [review]
LGTM lets do the < 23 later

**Pull request merged**